docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,457 @@
+verifiable credential and presentation described in the [document]().
+
+### Goals and ideas
+
+* Use Indy styled credentials to generate W3C AnonCreds credentials and presentations
+    * Credentials conversion:
+        * Convert Indy styled AnonCreds credentials into W3C form
+        * Convert W3C styled AnonCreds credentials into Indy form
+    * Presentation conversion (Optional):
+        * Convert Indy styled AnonCreds presentation into W3C form
+        * Convert W3C styled AnonCreds presentation into Indy form
+* Extend W3C credentials:
+    * Ability to set Data Integrity proof signatures for generated W3C credential objects:
+        * W3C credentials may contain multiple signatures
+        * AnonCreds-Rs only generates/handle AnonCreds signatures
+
+#### Out of scope
+
+* Credentials: Verify validity of the Data Integrity proof signature
+* Presentations: Create presentation using Data Integrity proof signature
+* Presentations: Verify validity of presentations using Data Integrity proof signatures
+* Presentations: Support different formats (for example DIF) of Presentation Request
+
+### Question impacting the approach
+
+* Q1: Do we need conversion for intermediate messages to make them W3C compliant: Credential Offer, Credential Request?
+    * Q1.1: (Depends on answer for Q1) If no conversion: Do we want Credential Offer indicates what form of credential
+      will be issued as the process execution result?
+    * Proposed answer: no conversion for the current phase
+* Q2: Do we want duplicate methods (like `sign` and `sign_w3c`) or only use single conversion method (
+  like `credential.to_w3c`) doing extra step?
+    * There are 6 methods in total. 4 of them we have to duplicate any way. Whether we want to
+      duplicate `create_offer`, `create_request` methods if we do not change their format.
+    * Proposed answer: yes - duplicate all methods
+* Q3: Are we still tied to Indy styled presentation request?
+    * Can we make interface completely independent of Presentation Request? So any form can be handled on top level and
+      specific data passed to AnonCreds-rs.
+    * Proposed answer: Make AnonCreds-rs methods presentation agnostic.
+* Q4: Do we want to provide general interfaces doing signing and verification of Data Integrity proof signature?
+    * Accept sign/verify callbacks into convert/create functions:
+        * Issue with setting multiple signatures
+        * Much easier just to expose methods to add signature proof itself
+    * Provide methods to put ready data into a credential
+    * Proposed answer: delegate to third party libraries using `anoncreds-rs`
+* Q5: Signature/Proof encoding: Which approach to use for encoding?
+    * Basic approach used in Aries attachments - Base 64 encoding of serialized
+      object (`base64(json(signature).as_bytes())`) ?
+    * Compact encoding implementing in Python PoC?
+    * Proposed answer: Start from basic approach?
```

Comment:
Do we have any idea of the size benefit?  Would a quick experiment help?  What did @andrewwhitehead do in the Python POC — https://github.com/andrewwhitehead/anoncreds-w3c-mapping?

Reply:
Please have a look at https://github.com/hyperledger/anoncreds-rs/blob/9e7763cd66c9a819ad7852f88b31912d853d81a1/docs/design/w3c/encoding.md#example

Reply:
Discussed during the Aries Announced community call. 
Decided to start with the base base64-url-safe encoding and reach other encoding algorithms as the next step.